UI/GameScreen.cpp

```diff
@@ -178,7 +179,8 @@ void GameScreen::CreateViews() {
 	btnSetBackground_->OnClick.Handle(this, &GameScreen::OnSetBackground);
 	btnSetBackground_->SetVisibility(V_GONE);
 
-	if (!Reporting::HasCRC(gamePath_)) {
+	bool fileTypeSupportCRC = info && (info->fileType == IdentifiedFileType::PSP_PBP || info->fileType == IdentifiedFileType::PSP_ISO_NP || info->fileType == IdentifiedFileType::PSP_ISO);
```

Comment:
Hm, for PSP_PBP_DIRECTORY, PSN games can have a CRC but homebrew can't always.

`InitMemoryForGamePBP` has some logic for determining if something is a homebrew, maybe we should refactor it so we can reuse.

-[Unknown]

Reply:
I don't think we can get a CRC from a PSP_PBP_DIRECTORY anyway can we? Calculating the CRC read the gamePath with this:
https://github.com/hrydgard/ppsspp/blob/ae9bf279910dc380183168f340665587a020b1d9/Core/FileLoaders/LocalFileLoader.cpp#L184

pread doesn't work on directory and will return -1, CRC will "detect" the fail and result in CRC = 0. I don't have any PSN game tho' so I don't really get what you mean tbh.

Reply:
For a PBP directory, it actually opens the EBOOT.PBP.  But it would actually route to `BlockDevice::CalculateCRC`, which uses `NPDRMDemoBlockDevice::ReadBlock`.  It would not use pread, as that would of course fail to calculate the correct CRC for CSO or PBP files.

PBP can be used as, essentially, a compressed ISO.

See `LoadFile()` in Loaders.cpp.  PSP_PBP_DIRECTORY turns into the actual PBP via ResolveFileLoaderTarget, which in turn reads as a PSP_ISO_NP.  We can still calculate a CRC in that case.

-[Unknown]

Reply:
I'm still kinda confused to be quite honest, sorry to be annoying...

It's true that `LoadFile` call `ResolveFileLoaderTarget`, but `LoadFile` get called by `CPU_Init`, here the CRC get calculated from the `GameScreen` AKA hold on the icon.

CRC button calls:
`Reporting::QueueCRC(gamePath_);`

The CRC calculating thread will then do (with `crcFilename = gamePath_`):
```
FileLoader *fileLoader = ConstructFileLoader(crcFilename);
BlockDevice *blockDevice = constructBlockDevice(fileLoader);
```

`ConstructFileLoader` should hit this (as is not HTTP and `RegisterFileLoaderFactory` is never called anywhere it seems):
`return new LocalFileLoader(filename);`

Now `constructBlockDevice` try to detect file type reading in it:
```
char buffer[4]{};
size_t size = fileLoader->ReadAt(0, 1, 4, buffer);
```

Now at least in Cave Story homebrew (that is detected as PBP directory and it seem so) it try to read a folder, resulting in `size = -1` (well not really being a `size_t`, but still) and so it wont use `NPDRMDemoBlockDevice`:
```
if (size == 4 && !memcmp(buffer, "CISO", 4))
	return new CISOFileBlockDevice(fileLoader);
else if (size == 4 && !memcmp(buffer, "\x00PBP", 4))
	return new NPDRMDemoBlockDevice(fileLoader);
else
	return new FileBlockDevice(fileLoader);
```

It will use `FileBlockDevice` which will try to read a folder and fail again resulting in the CRC being 0.
Does this work for real PBP from PSN? I'm missing something I guess...

Anyway I tried to use the ResolveFileTarget there too but with that cave story it end up in (I might have just failed somewhere tho')
```
51:52:721 FileSystems/BlockDevices.cpp:395 E[LOADER]: Invalid NPUMDIMG header!
Floating point exception (core dumped)
```

On side note tho' `GameInfoCache` update the region for homebrew like this:

```
// Assuming PSP_PBP_DIRECTORY without ID or with disc_total < 1 in GAME dir must be homebrew
if ((info_->id.empty() || !info_->disc_total)
	&& gamePath_.FilePathContains("PSP/GAME/")
	&& info_->fileType == IdentifiedFileType::PSP_PBP_DIRECTORY) {
	info_->id = g_paramSFO.GenerateFakeID(gamePath_.ToString());
	info_->id_version = info_->id + "_1.00";
	info_->region = GAMEREGION_MAX + 1; // Homebrew
}
```

So for homebrew we can just check the region? (well kinda euristic, but still).

Reply:
Ah, sorry. So the separate bug here is that this button doesn't currently work for NPUMDIMG (PSN ISO-as-PBP) but could.  I thought it used to for reporting... should be a quick fix, I'll update.

Maybe it'd work to check the region.  I'll make it at least not crash.

-[Unknown]

Reply:
See #14801, thought that was using the loaded file loader, but of course it's not since it can be triggered from outside a game now.

-[Unknown]

Reply:
Ok! Updated to work on PBP folder and to check homebrew with region (it avoid GE dump with the file type check as that CRC is meaningless).